File: android/src/main/kotlin/com/vn/th_barcode_scanner/helper/EventHandler.kt
package com.vn.th_barcode_scanner.helper

import io.flutter.plugin.common.EventChannel

class EventHandler : EventChannel.StreamHandler {
    internal var sink: EventChannel.EventSink? = null

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        sink = events
    }

    override fun onCancel(arguments: Any?) {

    }
}

File: ios/Classes/Helper/EventHandler.swift
//
//  EventHandler.swift
//  th_barcode_scanner
//
//  Created by Tuấn Hwing on 8/21/21.
//

import UIKit

class EventHandler: NSObject, FlutterStreamHandler {
    var sink:FlutterEventSink?
                
        func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
            self.sink = events
            return nil
        }
        
        func onCancel(withArguments arguments: Any?) -> FlutterError? {
            return nil
        }
}
